##################  Download public data_Establishment of Porcine and Human Expanded Potential Stem Cells(https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-7253/)
# ERR3164073, ERR3164074 > HPSI1014i-babz-3 (hiEPSC-TSC)
# ERR3164071, ERR3164072 > C5 (hEPSC-TSC)

# wget "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR316/003/ERR3164073/ERR3164073_1.fastq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR316/003/ERR3164073/ERR3164073_2.fastq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR316/004/ERR3164074/ERR3164074_1.fastq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR316/004/ERR3164074/ERR3164074_2.fastq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR316/001/ERR3164071/ERR3164071_1.fastq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR316/001/ERR3164071/ERR3164071_2.fastq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR316/002/ERR3164072/ERR3164072_1.fastq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR316/002/ERR3164072/ERR3164072_2.fastq.gz";

##################  Download public data_RNA-seq reveals conservation of function among the yolk sacs of human, mouse, and chicken (http://www.ebi.ac.uk/ena/data/view/PRJEB18767)
# human yolk sac
#run_accession	sample_accession	scientific_name	submitted_ftp
#ERR1773615	SAMEA39827668	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773615/SLX-8764.A008.C4EYHANXX.s_7.r_1.fq.gz
#ERR1773616	SAMEA39827668	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773616/SLX-8767.A008.C5UP5ANXX.s_3.r_1.fq.gz
#ERR1773617	SAMEA39827668	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773617/SLX-9611.A008.C6H2YANXX.s_7.r_1.fq.gz

#ERR1773618	SAMEA39828418	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773618/SLX-8764.A009.C4EYHANXX.s_7.r_1.fq.gz
#ERR1773619	SAMEA39828418	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773619/SLX-8767.A009.C5UP5ANXX.s_3.r_1.fq.gz
#ERR1773620	SAMEA39828418	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773620/SLX-9613.A009.C66TRANXX.s_8.r_1.fq.gz
#ERR1773621	SAMEA39828418	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773621/SLX-9615.A009.C66Y8ANXX.s_5.r_1.fq.gz

#ERR1773622	SAMEA39829168	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773622/SLX-8764.A011.C4EYHANXX.s_7.r_1.fq.gz
#ERR1773623	SAMEA39829168	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773623/SLX-8767.A011.C5UP5ANXX.s_3.r_1.fq.gz
#ERR1773624	SAMEA39829168	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773624/SLX-9611.A011.C6H2YANXX.s_7.r_1.fq.gz

#ERR1773625	SAMEA39829918	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773625/SLX-8764.A020.C4EYHANXX.s_7.r_1.fq.gz
#ERR1773626	SAMEA39829918	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773626/SLX-8767.A020.C5UP5ANXX.s_3.r_1.fq.gz
#ERR1773627	SAMEA39829918	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773627/SLX-9611.A020.C6H2YANXX.s_7.r_1.fq.gz
#ERR1773628	SAMEA39829918	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773628/SLX-9615.A020.C66Y8ANXX.s_5.r_1.fq.gz

#ERR1773629	SAMEA39830668	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773629/SLX-8764.A021.C4EYHANXX.s_7.r_1.fq.gz

#ERR1773630	SAMEA39831418	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773630/SLX-8764.A022.C4EYHANXX.s_7.r_1.fq.gz
#ERR1773631	SAMEA39831418	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773631/SLX-8767.A022.C5UP5ANXX.s_3.r_1.fq.gz

#ERR1773632	SAMEA39832168	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773632/SLX-8764.A023.C4EYHANXX.s_7.r_1.fq.gz
#ERR1773633	SAMEA39832168	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773633/SLX-8767.A023.C5UP5ANXX.s_3.r_1.fq.gz
#ERR1773634	SAMEA39832168	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773634/SLX-9613.A023.C66TRANXX.s_8.r_1.fq.gz
#ERR1773635	SAMEA39832168	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773635/SLX-9615.A023.C66Y8ANXX.s_5.r_1.fq.gz

#ERR1773636	SAMEA39832918	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773636/SLX-8764.A025.C4EYHANXX.s_7.r_1.fq.gz
#ERR1773637	SAMEA39832918	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773637/SLX-8767.A025.C5UP5ANXX.s_3.r_1.fq.gz

#ERR1773638	SAMEA39833668	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773638/SLX-8764.A027.C4EYHANXX.s_7.r_1.fq.gz
#ERR1773639	SAMEA39833668	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773639/SLX-8767.A027.C5UP5ANXX.s_3.r_1.fq.gz
#ERR1773640	SAMEA39833668	Homo sapiens	ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773640/SLX-9611.A027.C6H2YANXX.s_7.r_1.fq.gz

# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773615/SLX-8764.A008.C4EYHANXX.s_7.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773616/SLX-8767.A008.C5UP5ANXX.s_3.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773617/SLX-9611.A008.C6H2YANXX.s_7.r_1.fq.gz";

# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773618/SLX-8764.A009.C4EYHANXX.s_7.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773619/SLX-8767.A009.C5UP5ANXX.s_3.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773620/SLX-9613.A009.C66TRANXX.s_8.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773621/SLX-9615.A009.C66Y8ANXX.s_5.r_1.fq.gz";

# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773622/SLX-8764.A011.C4EYHANXX.s_7.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773623/SLX-8767.A011.C5UP5ANXX.s_3.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773624/SLX-9611.A011.C6H2YANXX.s_7.r_1.fq.gz";

# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773625/SLX-8764.A020.C4EYHANXX.s_7.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773626/SLX-8767.A020.C5UP5ANXX.s_3.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773627/SLX-9611.A020.C6H2YANXX.s_7.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773628/SLX-9615.A020.C66Y8ANXX.s_5.r_1.fq.gz";

# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773630/SLX-8764.A022.C4EYHANXX.s_7.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773631/SLX-8767.A022.C5UP5ANXX.s_3.r_1.fq.gz";

# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773636/SLX-8764.A025.C4EYHANXX.s_7.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773637/SLX-8767.A025.C5UP5ANXX.s_3.r_1.fq.gz";

# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773638/SLX-8764.A027.C4EYHANXX.s_7.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773639/SLX-8767.A027.C5UP5ANXX.s_3.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773640/SLX-9611.A027.C6H2YANXX.s_7.r_1.fq.gz";

# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773632/SLX-8764.A023.C4EYHANXX.s_7.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773633/SLX-8767.A023.C5UP5ANXX.s_3.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773634/SLX-9613.A023.C66TRANXX.s_8.r_1.fq.gz";
# wget "ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR177/ERR1773635/SLX-9615.A023.C66Y8ANXX.s_5.r_1.fq.gz";


##################  samples combine
cat *A008* > humanYolksac_RNAseq_1.fastq.gz;
cat *A009* > humanYolksac_RNAseq_2.fastq.gz;
cat *A011* > humanYolksac_RNAseq_3.fastq.gz;
cat *A020* > humanYolksac_RNAseq_4.fastq.gz;
cat *A021* > humanYolksac_RNAseq_5.fastq.gz;
cat *A022* > humanYolksac_RNAseq_6.fastq.gz;
cat *A023* > humanYolksac_RNAseq_7.fastq.gz;
cat *A025* > humanYolksac_RNAseq_8.fastq.gz;
cat *A027* > humanYolksac_RNAseq_9.fastq.gz;

##################  Align to Gene with STAR (run in Iyer server:path="/stor/work/JKim/Yu/TSLC-RNAseq/fq")
############################################################################################
##########################       RNA-seq analysis (hg38)    ################################
############################################################################################
##################  Genome generation (for indexing)
# https://physiology.med.cornell.edu/faculty/skrabanek/lab/angsd/lecture_notes/STARmanual.pdf

path="/stor/work/JKim/Yu/TSLC-RNAseq/fq"
path1="/stor/work/JKim/Yu/TSLC-RNAseq/STAR"

##################  Mapping (run in adler3:/stor/work/JKim/MK/RNA-seq/JA21134_EVT_KD)
# STAR --runThreadN 20 --genomeDir /stor/work/JKim/MK/Genome/STAR_Human --readFilesIn humanYolksac_RNAseq_1.fastq --outFileNamePrefix /stor/work/JKim/Yu/TSLC-RNAseq/STAR/humanYolksac_1;
# STAR --runThreadN 20 --genomeDir /stor/work/JKim/MK/Genome/STAR_Human --readFilesIn humanYolksac_RNAseq_2.fastq --outFileNamePrefix /stor/work/JKim/Yu/TSLC-RNAseq/STAR/humanYolksac_2;
# STAR --runThreadN 20 --genomeDir /stor/work/JKim/MK/Genome/STAR_Human --readFilesIn humanYolksac_RNAseq_3.fastq --outFileNamePrefix /stor/work/JKim/Yu/TSLC-RNAseq/STAR/humanYolksac_3;
# STAR --runThreadN 20 --genomeDir /stor/work/JKim/MK/Genome/STAR_Human --readFilesIn humanYolksac_RNAseq_4.fastq --outFileNamePrefix /stor/work/JKim/Yu/TSLC-RNAseq/STAR/humanYolksac_4;
# STAR --runThreadN 20 --genomeDir /stor/work/JKim/MK/Genome/STAR_Human --readFilesIn humanYolksac_RNAseq_5.fastq --outFileNamePrefix /stor/work/JKim/Yu/TSLC-RNAseq/STAR/humanYolksac_5;
# STAR --runThreadN 20 --genomeDir /stor/work/JKim/MK/Genome/STAR_Human --readFilesIn humanYolksac_RNAseq_6.fastq --outFileNamePrefix /stor/work/JKim/Yu/TSLC-RNAseq/STAR/humanYolksac_6;
# STAR --runThreadN 20 --genomeDir /stor/work/JKim/MK/Genome/STAR_Human --readFilesIn humanYolksac_RNAseq_7.fastq --outFileNamePrefix /stor/work/JKim/Yu/TSLC-RNAseq/STAR/humanYolksac_7;
# STAR --runThreadN 20 --genomeDir /stor/work/JKim/MK/Genome/STAR_Human --readFilesIn humanYolksac_RNAseq_8.fastq --outFileNamePrefix /stor/work/JKim/Yu/TSLC-RNAseq/STAR/humanYolksac_8;
# STAR --runThreadN 20 --genomeDir /stor/work/JKim/MK/Genome/STAR_Human --readFilesIn humanYolksac_RNAseq_9.fastq --outFileNamePrefix /stor/work/JKim/Yu/TSLC-RNAseq/STAR/humanYolksac_9;

# STAR --runThreadN 20 --genomeDir /stor/work/JKim/MK/Genome/STAR_Human --readFilesIn $path/ERR3164073_1.fastq $path/ERR3164073_2.fastq --outFileNamePrefix $path1/hiEPSC-TSC_1;
# STAR --runThreadN 20 --genomeDir /stor/work/JKim/MK/Genome/STAR_Human --readFilesIn $path/ERR3164074_1.fastq $path/ERR3164074_2.fastq --outFileNamePrefix $path1/hiEPSC-TSC_2;
# STAR --runThreadN 20 --genomeDir /stor/work/JKim/MK/Genome/STAR_Human --readFilesIn $path/ERR3164071_1.fastq $path/ERR3164071_2.fastq --outFileNamePrefix $path1/hEPSC-TSC_1;
# STAR --runThreadN 20 --genomeDir /stor/work/JKim/MK/Genome/STAR_Human --readFilesIn $path/ERR3164072_1.fastq $path/ERR3164072_2.fastq --outFileNamePrefix $path1/hEPSC-TSC_2;

##################  Quantification with HTseq (run in adler3:/stor/work/JKim/Yu/TSLC-RNAseq/fq)
## The chr name should be matched with sam file format. chr1 = chr1, not this chr1 = 1.
## gencode.vM20.chr_patch_hapl_scaff.annotation.gtf : evidence-based annotation of the mouse genome (GRCm38), version M20 (Ensembl 95)

path2="/stor/work/JKim/Yu/TSLC-RNAseq"
path3="/stor/work/JKim/Yu/TSLC-RNAseq/HTseq"

htseq-count -t exon $path1/hEPSC-TSC_1Aligned.out.sam $path2/Homo_sapiens_genecode.v35.annotation.gtf > $path3/hEPSC-TSC_1_HTseq.counts
htseq-count -t exon $path1/hEPSC-TSC_2Aligned.out.sam $path2/Homo_sapiens_genecode.v35.annotation.gtf > $path3/hEPSC-TSC_2_HTseq.counts
htseq-count -t exon $path1/hiEPSC-TSC_1Aligned.out.sam $path2/Homo_sapiens_genecode.v35.annotation.gtf > $path3/hiEPSC-TSC_1_HTseq.counts
htseq-count -t exon $path1/hiEPSC-TSC_2Aligned.out.sam $path2/Homo_sapiens_genecode.v35.annotation.gtf > $path3/hiEPSC-TSC_2_HTseq.counts
htseq-count -t exon $path1/humanYolksac_1Aligned.out.sam $path2/Homo_sapiens_genecode.v35.annotation.gtf > $path3/humanYolksac_1_HTseq.counts
htseq-count -t exon $path1/humanYolksac_2Aligned.out.sam $path2/Homo_sapiens_genecode.v35.annotation.gtf > $path3/humanYolksac_2_HTseq.counts
htseq-count -t exon $path1/humanYolksac_3Aligned.out.sam $path2/Homo_sapiens_genecode.v35.annotation.gtf > $path3/humanYolksac_3_HTseq.counts
htseq-count -t exon $path1/humanYolksac_4Aligned.out.sam $path2/Homo_sapiens_genecode.v35.annotation.gtf > $path3/humanYolksac_4_HTseq.counts
htseq-count -t exon $path1/humanYolksac_5Aligned.out.sam $path2/Homo_sapiens_genecode.v35.annotation.gtf > $path3/humanYolksac_5_HTseq.counts
htseq-count -t exon $path1/humanYolksac_6Aligned.out.sam $path2/Homo_sapiens_genecode.v35.annotation.gtf > $path3/humanYolksac_6_HTseq.counts
htseq-count -t exon $path1/humanYolksac_7Aligned.out.sam $path2/Homo_sapiens_genecode.v35.annotation.gtf > $path3/humanYolksac_7_HTseq.counts
htseq-count -t exon $path1/humanYolksac_8Aligned.out.sam $path2/Homo_sapiens_genecode.v35.annotation.gtf > $path3/humanYolksac_8_HTseq.counts
htseq-count -t exon $path1/humanYolksac_9Aligned.out.sam $path2/Homo_sapiens_genecode.v35.annotation.gtf > $path3/humanYolksac_9_HTseq.counts
